contracts/OddsLib.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.10;
+
+import "@openzeppelin/contracts/utils/math/Math.sol";
+
+library OddsLib {
+
+    uint256 public constant PRECISION = 1e6;
+
+    /*
+        * @dev Reduces odds linearly to 0 (when the payout would exceed the free liquidity)
+        * @param wager The amount of the wager
+        * @param odds The odds of the wager, expressed as a number from 0 to PRECISION
+        * @return The new adjusted odds
+    */
+    function getLinearAdjustedOdds(
+        uint256 wager,
+        uint256 odds,
+        uint256 maxPayout
+    ) external pure returns (uint256) {
+        uint256 unadjustedPayout = odds * wager / PRECISION;
+		uint256 oddsAdjustment = Math.mulDiv(odds, unadjustedPayout, maxPayout, Math.Rounding.Up);
+        // If we have gone past the floor, clip to 0
+        if (oddsAdjustment > odds) {
+            return 0;
+        }
+        return odds - oddsAdjustment;
+    }
```

Comment:
I dont agree with this.  The odds have to slip as the bet increases to the size of the pool.